src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts

```diff
@@ -26,9 +26,16 @@ export class FormatterService {
     if (!_.isNumber(n)) {
       return '-';
     }
-    const unit = n < 1 ? 0 : Math.floor(Math.log(n) / Math.log(divisor));
-    const truncatedFloat = this.truncate(n / Math.pow(divisor, unit), decimals);
-    return truncatedFloat === '' ? '-' : truncatedFloat + units[unit];
+    let unit = n < 1 ? 0 : Math.floor(Math.log(n) / Math.log(divisor));
+    unit = (unit >= units.length) ? units.length - 1 : unit;
+    let result = this.truncate(n / Math.pow(divisor, unit), decimals);
+    if (result === '') {
+      return '-';
+    }
+    if (units[unit] !== '') {
```

Comment:
I think you can now remove this line :)
This said how about converting it to a one liner :D 
```
return result === '' ? '-' : `${result} ${units[unit]}`
```

Reply:
We still need that line because the list of units can contain empty strings, see https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/dimless.pipe.ts#L12.

Reply:
I know this is merged, but the empty string shouldn't be empty it should be `b` instead.